app/controllers/users/sessions_controller.rb

```diff
@@ -115,6 +115,7 @@ def process_locked_out_user
     def handle_valid_authentication
       sign_in(resource_name, resource)
       cache_profiles(auth_params[:password])
+      user_session[:new_device] = current_user.new_device?(device_cookie: cookies[:device])
       create_user_event(:sign_in_before_2fa)
```

Comment:
Should we update the logic in `create_user_event` to use the new `new_device?` method?

https://github.com/18F/identity-idp/blob/61c5e6685f87ee383941d4924ecabd1fab823513/app/services/user_event_creator.rb#L14-L18

Reply:
That returns a boolean but this line is looking for the device object
https://github.com/18F/identity-idp/blob/61c5e6685f87ee383941d4924ecabd1fab823513/app/services/user_event_creator.rb#L20

The method would need to return the object and the places it's referenced would need to have .present? I think.

Reply:
> That returns a boolean but this line is looking for the device object

Ah yeah, that's a good point. Maybe it's fine to leave as-is since I wouldn't want to go overboard with abstractions for what's ultimately pretty simple logic.